File: XiaMenOA/src/main/java/com/itheima/service/AssetService.java
package com.itheima.service;

import com.itheima.domain.TbAsset;
import com.itheima.domain.TbAssetType;
import com.itheima.utils.AssetMsgs;
import com.itheima.utils.AssetSearch;
import com.itheima.utils.PageResult;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;

public interface AssetService {
    PageResult findByPage();

    void fixedAssetImport(MultipartFile excelFile) throws IOException;

    void lowAssetImport(MultipartFile excelFile) throws IOException;

    void assetTypeImport(MultipartFile excelFile) throws IOException;

    List findAssetMsgs(List<TbAsset> list);

    PageResult search(AssetSearch assetSearch, List<Integer> idList);

    List<Integer> selectAllByParentId(String typeName);

    List<TbAssetType> selectAllByIdList(List<Integer> idList);

    List<TbAssetType> selectAllByIdList(List<TbAssetType> secondList, Integer secondTypeName);

    void assetAdd(TbAsset asset);

    TbAsset findOne(Integer id);

    AssetMsgs findAssetMsgs(TbAsset asset);

    void del(Integer id, Integer useableFlag);
}

File: XiaMenOA/src/main/webapp/js/service/assetService.js
//服务层
app=angular.module("XiaMenOA",["pagination"]).service('assetService',function($http){

	//分页 
	this.findPage=function(page,rows){
		return $http.get('../asset/findByPage.do?page='+page+'&rows='+rows);
	}
    //搜索
    this.search=function(page,rows,searchPage){
        return $http.post('../asset/search.do?page='+page+"&rows="+rows, searchPage);
    }
    //查找单个资产信息
    this.assetEdit=function (id) {
        return $http.get('../asset/findOne.do?id='+id);
    }

    //删除单个资产信息
    this.asset_Delete=function (id,useableFlag) {
        return $http.get('../asset/del.do?id='+id+"&useableFlag="+useableFlag);
    }
});
